### PR TITLE
reports: fix missing alert details in Risk and Confidence HTML

### PR DESCRIPTION
**Pull request by GunoH**
## Overview
Fixes an issue where alert details were missing from some reports.

## Related Issues
Fix zaproxy/zaproxy#8460.

## Checklist
- [x] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [x] Write tests
- [x] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

For more details, please refer to the [developer rules and guidelines](https://www.zaproxy.org/docs/developer/dev-rules-and-guidelines/).

### CODE REVIEW

**Comment by GunoH** [comment]
I have read the CLA Document and I hereby sign the CLA

File: addOns/reports/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ### Changed
 - Checkmarx rebrand.
 
+### Fixed
+- Fixed an issue where alert details were missing from some reports.
```
Comment (kingthorin) [inline addOns/reports/CHANGELOG.md:11]: > - An issue where alert details were missing from some reports (Issue 8460).

File: addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportHelper.java [hunk 2]
```diff
@@ -73,18 +75,27 @@ public static int getPortForSite(String site) {
...
-                // Ignore
+
+        try {
+            var url = new URL(site);
+            var port = url.getPort();
+            if (port != -1) {
+                return port;
+            }
+
+            if (url.getProtocol().equals("https")) {
```
Comment (kingthorin) [inline addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportHelper.java:86]: This should probably use an ignorecase variant, which may mean using StringUtils

File: addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportHelper.java [hunk 3]
```diff
@@ -73,18 +75,27 @@ public static int getPortForSite(String site) {
...
+                return 443;
+            } else {
+                return 80;
+            }
+
+        } catch (MalformedURLException e) {
+            if (site.startsWith("https")) {
```
Comment (kingthorin) [inline addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportHelper.java:93]: This should probably use an ignorecase variant, which may mean using StringUtils

File: addOns/reports/CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ### Changed
 - Checkmarx rebrand.
 
+### Fixed
+- An issue where alert details were missing from some reports (Issue 8460).
```
Comment (thc202) [inline addOns/reports/CHANGELOG.md:11]: `…from Risk and Confidence HTML reports…`.

File: addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportHelper.java [hunk 5]
```diff
@@ -73,18 +76,27 @@ public static int getPortForSite(String site) {
...
         }
-        String[] schemeHostPort = site.split(":");
-        if (schemeHostPort.length == 3) {
-            try {
-                return Integer.parseInt(schemeHostPort[2]);
-            } catch (NumberFormatException e) {
-                // Ignore
+
+        try {
+            var url = new URL(site);
```
Comment (thc202) [inline addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportHelper.java:81]: Use `URI` class.

File: addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportHelper.java [hunk 6]
```diff
@@ -73,18 +76,27 @@ public static int getPortForSite(String site) {
...
+            var port = url.getPort();
```
Comment (thc202) [inline addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportHelper.java:82]: `int port`

File: addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportHelper.java [hunk 7]
```diff
@@ -73,18 +76,27 @@ public static int getPortForSite(String site) {
...
+            if (port != -1) {
+                return port;
+            }
+
+            if (url.getProtocol().equalsIgnoreCase("https")) {
+                return 443;
+            } else {
+                return 80;
+            }
```
Comment (thc202) [inline addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportHelper.java:91]: This should be deduplicated.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]

**Pull request merged**